### PR TITLE
Maven: extract property and get version

### PR DESCRIPTION
**Pull request by DanielFran**
Related with https://github.com/jhipster/jhipster-lite/issues/278

### CODE REVIEW

**Comment by DanielFran** [comment]
@pascalgrimaud Is it OK for you?

**Comment by pascalgrimaud** [comment]
Yes ping me when it's ready for review

File: src/main/java/tech/jhipster/lite/generator/buildtool/generic/domain/BuildToolDomainService.java [hunk 1]
```diff
@@ -106,4 +107,13 @@ public void init(Project project, BuildToolType buildTool) {
       throw new GeneratorException(EXCEPTION_NO_BUILD_TOOL);
     }
   }
+
+  @Override
+  public Optional<String> getVersion(Project project, String name) {
+    if (project.isMavenProject()) {
```
Comment (qmonmert) [inline src/main/java/tech/jhipster/lite/generator/buildtool/generic/domain/BuildToolDomainService.java:113]: you can remove the `else`

```
if (project.isMavenProject()) {
    return mavenService.getVersion(name);
} 
throw new GeneratorException(EXCEPTION_NO_BUILD_TOOL);
```

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: src/main/java/tech/jhipster/lite/generator/buildtool/maven/domain/MavenDomainService.java [hunk 2]
```diff
@@ -240,4 +244,23 @@ public void addMavenWrapper(Project project) {
     projectRepository.add(project, sourceWrapper, "maven-wrapper.jar", destinationWrapper);
     projectRepository.add(project, sourceWrapper, "maven-wrapper.properties", destinationWrapper);
   }
+
+  @Override
+  public Optional<String> getVersion(String name) {
+    Assert.notBlank("name", name);
+
+    String property_tag_ini = new StringBuilder().append("<").append(name).append(".version").append(">").toString();
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/buildtool/maven/domain/MavenDomainService.java:252]: sonar code smell: use `propertyTagIni` instead

File: src/main/java/tech/jhipster/lite/generator/buildtool/maven/domain/MavenDomainService.java [hunk 3]
```diff
@@ -240,4 +244,23 @@ public void addMavenWrapper(Project project) {
...
+    String property_tag_end = new StringBuilder().append("</").append(name).append(".version").append(">").toString();
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/buildtool/maven/domain/MavenDomainService.java:253]: use `propertyTagEnd` instead

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: src/test/java/tech/jhipster/lite/generator/buildtool/maven/domain/maven/MavenDomainServiceTest.java [hunk 4]
```diff
@@ -181,4 +188,44 @@ void shouldAddMavenWrapper() {
...
+
+    @Test
+    void shouldNotGetVersion() {
+      assertThat(mavenDomainService.getVersion("unknown")).isEmpty();
+    }
+
+    @Test
+    void shouldNotGetVersionForWrongPattern() {
+      try (MockedStatic<FileUtils> fileUtilsMock = Mockito.mockStatic(FileUtils.class)) {
+        fileUtilsMock.when(() -> FileUtils.getPath(Mockito.anyString())).thenReturn("mypath");
```
[the remark this answers is not in the document]
Comment (DanielFran) [reply src/test/java/tech/jhipster/lite/generator/buildtool/maven/domain/maven/MavenDomainServiceTest.java:222]: Forget to fix the test, sorry.

**Review by pascalgrimaud — APPROVED** [review]

File: src/main/java/tech/jhipster/lite/generator/buildtool/generic/domain/BuildToolDomainService.java (hunk 1 above)
Comment (pascalgrimaud) [reply src/main/java/tech/jhipster/lite/generator/buildtool/generic/domain/BuildToolDomainService.java:113]: @DanielFran : can you do this minor change plz?

**Comment by DanielFran** [comment]
@pascalgrimaud anything missing?

**Comment by pascalgrimaud** [comment]
@DanielFran : yes, last comment from @qmonmert

image

**Review by qmonmert — APPROVED** [review]

**Pull request merged**